subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -0,0 +1,1020 @@
+[cols="a,a,a", options="header"]
+|===
+| Task Property
+| Type
+| Default Value
+
+| `srcDirs`
+| `Object`. Can set using anything described in <<sec:specifying_multiple_files>>.
+| `__sourceSet__.resources`
+
+| `destinationDir`
+| `File`. Can set using anything described in <<sec:locating_files>>.
+| `__sourceSet__.output.resourcesDir`
+|===
+
+
+[[sec:compile]]
+=== CompileJava
+
+The Java plugin adds a api:org.gradle.api.tasks.compile.JavaCompile[] instance for each source set in the project. Some of the most common configuration options are shown below.
+
+.Java plugin - Compile properties
+[cols="a,a,a", options="header"]
+|===
+| Task Property
+| Type
+| Default Value
+
+| `classpath`
+| api:org.gradle.api.file.FileCollection[]
+| `__sourceSet__.compileClasspath`
+
+| `source`
+| api:org.gradle.api.file.FileTree[]. Can set using anything described in <<sec:specifying_multiple_files>>.
+| `__sourceSet__.java`
+
+| `destinationDir`
+| `File`.
+| `__sourceSet__.output.classesDir`
+|===
+
+By default, the Java compiler runs in the Gradle process. Setting `options.fork` to `true` causes compilation to occur in a separate process. In the case of the Ant javac task, this means that a new process will be forked for each compile task, which can slow down compilation. Conversely, Gradle's direct compiler integration (see above) will reuse the same compiler process as much as possible. In both cases, all fork options specified with `options.forkOptions` will be honored.
+
+[[sec:incremental_compile]]
+=== Incremental Java compilation
+
+Starting with Gradle 2.1, it is possible to compile Java incrementally. See the api:org.gradle.api.tasks.compile.JavaCompile[] task for information on how to enable it.
+
+Main goals for incremental compilations are:
+
+* Avoid wasting time compiling source classes that don't have to be compiled. This means faster builds, especially when a change to a source class or a jar does not incur recompilation of many source classes that depend on the changed input.
+* Change as few output classes as possible. Classes that don't need to be recompiled remain unchanged in the output directory. An example scenario when this is really useful is using JRebel - the fewer output classes are changed the quicker the JVM can use refreshed classes.
+
+The incremental compilation at a high level:
+
+* The detection of the correct set of stale classes is reliable at some expense of speed. The algorithm uses bytecode analysis and deals gracefully with compiler optimizations (inlining of non-private constants), transitive class dependencies, etc. Example: When a class with a public constant changes, we eagerly compile classes that use the same constants to avoid problems with constants inlined by the compiler.
+* To make incremental compilation fast, we cache class analysis results and jar snapshots. The initial incremental compilation can be slower due to the cold caches.
+
+[[sec:incremental_compilation_known_issues]]
+==== Known issues
```

Comment:
Seems like this section is numbered in the asciidoc version (46.12.1) while it has no number in the docbook only version.